src/Tag.js

```diff
@@ -129,12 +129,19 @@ class Tag {
         }
         var written = tagStream.size + 4;
 
+        const isBigExplicit =
+            vr.isExplicit() &&
+            !implicit &&
+            valueLength >= 0x10000 &&
+            valueLength !== 0xffffffff;
+
         if (implicit) {
             stream.writeUint32(valueLength);
             written += 4;
         } else {
-            if (vr.isExplicit()) {
-                stream.writeAsciiString(vr.type);
+            if (vr.isExplicit() || isBigExplicit) {
```

Comment:
And this change actually writes the UN instead.
Note the vr.isExplicit is mis-named - it SHOULD be isSeparateLength or something like that.  A vr can't be explicit, it can only have a separate length field or a combined length field.